java-frontend/src/main/java/org/sonar/java/model/LiteralUtils.java

```diff
@@ -114,10 +114,14 @@ public static String trimLongSuffix(String longString) {
   }
 
   public static boolean hasValue(Tree tree, String expectedValue) {
-    if (!tree.is(Tree.Kind.STRING_LITERAL)) {
+    if (!tree.is(Kind.STRING_LITERAL)) {
       return false;
     }
-    String mechanismName = trimQuotes(((LiteralTree) tree).value());
-    return expectedValue.equals(mechanismName);
+    String actualValue = trimQuotes(((LiteralTree) tree).value());
+    return expectedValue.equals(actualValue);
+  }
+
+  public static boolean hasValue(Tree tree, boolean expectedValue) {
```

Comment:
I would prefer to name this method "hasBooleanValue".

Reply:
I don't agree with that. It's clear it's a boolean from the parameter.